src/include/parser/expression/comparison_expression.h

```diff
@@ -48,7 +53,28 @@ class ComparisonExpression : public AbstractExpression {
     return expr;
   }
 
-  void Accept(SqlNodeVisitor *v, ParseResult *parse_result) override { v->Visit(this, parse_result); }
+  void Accept(SqlNodeVisitor *v, ParseResult *parse_result) override {
```

Comment:
I'm not happy with this hack, but this is both the fastest and cleanest way that I can see us reusing the binder infrastructure right now. A similar thing might need to be done for arithmetic expressions.

We'd ideally want to do this at the ConstantValueExpression containing the NULL itself, but by that point you've lost the context of "what type of null should this be?".